PhysicsTools/PatAlgos/plugins/MiniAODRefitVertexProducer.cc

```diff
@@ -0,0 +1,189 @@
+
+  This producer is intended to take packedCandidates with tracks associated to
+  the PV and refit the PV (applying or not) BeamSpot constraint
+  
+  \autor Michal Bluj, NCBJ Warsaw (and then others)
+
+ **/
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "RecoVertex/VertexPrimitives/interface/TransientVertex.h"
+#include "RecoVertex/AdaptiveVertexFit/interface/AdaptiveVertexFitter.h"
+
+#include <memory>
+
+class MiniAODRefitVertexProducer : public edm::stream::EDProducer<> {
+public:
+  explicit MiniAODRefitVertexProducer(const edm::ParameterSet&);
+  ~MiniAODRefitVertexProducer() override{};
+
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  //--- utility methods
+
+  //--- configuration parameters
+  edm::EDGetTokenT<std::vector<pat::PackedCandidate> > srcCands_, srcLostTracks_, srcEleKfTracks_;
+  edm::EDGetTokenT<reco::VertexCollection> srcVertices_;
+  edm::EDGetTokenT<reco::BeamSpot> srcBeamSpot_;
+  edm::ESGetToken<TransientTrackBuilder, TransientTrackRecord> transTrackBuilderToken_;
+  bool useBeamSpot_;
+  bool useLostTracks_;
+  bool useEleKfTracks_;
+};
+
+MiniAODRefitVertexProducer::MiniAODRefitVertexProducer(const edm::ParameterSet& cfg)
+    : srcCands_(consumes<std::vector<pat::PackedCandidate> >(cfg.getParameter<edm::InputTag>("srcCands"))),
+      srcLostTracks_(consumes<std::vector<pat::PackedCandidate> >(cfg.getParameter<edm::InputTag>("srcLostTracks"))),
+      srcEleKfTracks_(consumes<std::vector<pat::PackedCandidate> >(cfg.getParameter<edm::InputTag>("srcEleKfTracks"))),
+      srcVertices_(consumes<reco::VertexCollection>(cfg.getParameter<edm::InputTag>("srcVertices"))),
+      srcBeamSpot_(consumes<reco::BeamSpot>(cfg.getParameter<edm::InputTag>("srcBeamSpot"))),
+      transTrackBuilderToken_(esConsumes(edm::ESInputTag("", "TransientTrackBuilder"))),
+      useBeamSpot_(cfg.getParameter<bool>("useBeamSpot")),
+      useLostTracks_(cfg.getParameter<bool>("useLostTracks")),
+      useEleKfTracks_(cfg.getParameter<bool>("useEleKfTracks")) {
+  produces<reco::VertexCollection>();
+}
+
+void MiniAODRefitVertexProducer::produce(edm::Event& evt, const edm::EventSetup& es) {
```

Comment:
no need for MiniAOD in the module name : are you certain there does not exist one such module under RecoVertex ?

Reply:
alright, following previous logic, I think you need to change it to `PATRefitVertexProducer`

Reply:
OK